templates/patterns/list_item/pattern-list-item--variant-thumbnail-primary.html.twig

```diff
@@ -2,14 +2,44 @@
 /**
  * @file
  * Default theme implementation for a ECL thumbnail list item component.
+ *
+ * @see @ecl/content-item
  */
 #}
-{% include '@ecl/ec-component-list-item' with {
-  'variant': 'thumbnail',
-  'title': title,
-  'metas': meta,
-  'href': url,
-  'detail': detail,
-  'primary_image': image,
-  'screen_reader': 'List item'|t
+
+{% set _image_position = image_position|default('before') %}
+
+{% set _meta = '' %}
+{% for meta_item in meta %}
+  {% if _meta != '' %}
+    {% set _meta = _meta ~ ' | ' %}
+  {% endif %}
+  {% set _meta = _meta ~ meta_item %}
+{% endfor %}
+
+{% set _infos = {} %}
```

Comment:
This transformation happens more than once. Why not do it in the preprocess function so it applies to all variants regardless?

Reply:
It's true that this transformation happens more than once, but these patterns start to use different ways (Highline, navigation, thumbnail, others) so it's more flexible to leave it there.
And to be consistent with other pattern components, I prefer to leave this transformation in the template as pattern_page_header, pattern_file, others.
But I am open to change my opinion.